doc/schema/schema.md

```diff
@@ -40,10 +41,20 @@ Currently there are 2 beta implementations of this design:
  <li><a class="reference internal" href="#examples">Examples</a></li>
 </ol>
 
+## Terminology
+The terminology used in this specification is based on [xarray's terminology](http://xarray.pydata.org/en/stable/terminology.html), however, no xarray knowledge is assumed in this description. There are also some extensions particular to  the InferenceData case.
+
+* **Variable**: NetCDF-like variables are multidimensional labeled arrays representing a single quantity. Variables and their dimensions must be named. They can also have attributes describing it. Relevant terms related to InferenceData variables are: *variable_name*, *values* (its data), *dimensions*, *coordinates*, and *attributes*.
+* **Dimension**: The dimensions of an object are its named axes. A variable containing 3D data can have dimensions `[chain, draw, dim0]`, thus, its `0th`-dimension is `chain`, its `1st`-dimension is `draw` and so on. Every dimension present in an InferenceData variable must share names with a *coordinate*. Given that dimensions must be named, dimension and dimension name are used equivalents.
+* **Coordinate**: A named array that labels a dimension. A coordinate named `chain` with values `[0, 1, 2, 3]` would label the `chain` dimension. Coordinate names and values can be loosely though of as labels and tick labels along a dimension. 
+* **Group**: Dataset containing one or several variables with a conceptual link between them. Variables inside a group will generally share some dimensions too.
+* **Matching samples**: Two variables (or groups) whose samples match are the ones generated with the same set of samples. Therefore, they will share dimensions and coordinates relative to samples. Sample dimensions (generally `(chain, draw)`) are the ones introduced by the sampling process. 
+* **Matching variables**: Two groups with matching variables are groups that conceptually share variables, variable dimensions and coordinates of the variable dimensions but do not necessarily share variable names nor sample dimensions. Variable dimensions are the ones intrinsic to the data and model as opposed to sample dimensions which are the ones relative to the sampling process. When talking about specific variables, this same idea is expressed as one variable being the counterpart of the other. 
```

Comment:
I really didn't understand this.  What does it mean to conceptually share variables, if the variables can have different names?

This seems like an informal notion that's not capable of crisp definition, so maybe drop it?

Reply:
We can try to define this better in a following PR maybe. 

I am trying to include cases like PyStan, where the observed variable and the posterior predictive cannot have the same name. This is the reason for example that the argument `data_pairs` exists in `plot_ppc`. #843 may provide some context.